src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -0,0 +1,973 @@
+                        project.RemoveItem(unresolvedReferenceItem);
+                    }
+                }
+
+                // Handle the removal of shared import nodes.
+                var projectXml = await access.GetProjectXmlAsync(UnconfiguredProject.FullPath)
+                                             .ConfigureAwait(true);
+                foreach (var sharedImportNode in sharedImportNodes)
+                {
+                    // Find the import that is included in the evaluation of the specified ConfiguredProject that
+                    // imports the project file whose full path matches the specified one.
+                    var matchingImports = from import in project.Imports
+                                          where import.ImportingElement.ContainingProject == projectXml
+                                          where PathHelper.IsSamePath(import.ImportedProject.FullPath, 
+                                                                      sharedImportNode.FilePath)
+                                          select import;
+                    foreach (var importToRemove in matchingImports)
+                    {
+                        var importingElementToRemove = importToRemove.ImportingElement;
+                        Report.IfNot(importingElementToRemove != null, 
+                                     "Cannot find shared project reference to remove.");
+                        if (importingElementToRemove != null)
+                        {
+                            await access.CheckoutAsync(importingElementToRemove.ContainingProject.FullPath)
+                                        .ConfigureAwait(true);
+                            importingElementToRemove.Parent.RemoveChild(importingElementToRemove);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Finds dependencies child nodes by their path. We need to override it since
+        /// we need to find children under either:
+        ///     - our dependencies root node.
+        ///     - dependency sub tree nodes
+        ///     - dependency sub tree top level nodes
+        /// (deeper levels will be graph nodes with additional info, not direct dependencies
+        /// specified in the project file or project.json)
+        /// </summary>
+        public override IProjectTree FindByPath(IProjectTree root, string path)
+        {
+            var dependenciesNode = GetSubTreeRootNode(root, DependenciesRootNodeFlags);
+            if (dependenciesNode == null)
+            {
+                return null;
+            }
+
+            // Note: all dependency nodes file path starts with file:/// to make sure we have 
+            // valid absolute path everytime.
+            if (!path.StartsWith("file:///"))
+            {
+                // just in case if given path is not in uri format
+                path = "file:///" + path.Trim('/');
+            }
+
```

Comment:
No, helper should be generic, but this code is relying on specific node paths format etc 

---
In reply to: [74325407](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74325407) [](ancestors = 74325407)